centreon/www/include/monitoring/status/Services/xml/serviceXML.php

```diff
@@ -165,35 +165,62 @@
     h.icon_image AS h_icon_images, h.display_name AS h_display_name, h.action_url AS h_action_url,
     h.notes_url AS h_notes_url, h.notes AS h_notes, h.address,
     h.passive_checks AS h_passive_checks, h.active_checks AS h_active_checks,
-    i.name as instance_name, cv.value as criticality, cv.value IS NULL as isnull
-    FROM hosts h, instances i ";
+    i.name as instance_name, cv.value as criticality, cv.value IS NULL as isnull,
+    m.index_id IS NOT NULL as hasGraph
+    FROM hosts h
+    INNER JOIN instances i
+      ON h.instance_id = i.instance_id
+    INNER JOIN services s
+      ON s.host_id = h.host_id
+    LEFT JOIN index_data idx
+      ON idx.host_id = h.host_id
+      AND idx.service_id = s.service_id
+      AND idx.hidden = '0'
+    LEFT JOIN metrics m
+      ON m.index_id = idx.id ";
+
 if (isset($hostgroups) && $hostgroups != 0) {
-    $request .= ", hosts_hostgroups hg, hostgroups hg2";
+    $request .= "INNER JOIN hosts_hostgroups hg
+        ON hg.host_id = h.host_id
+        AND hg.hostgroup_id = :hostGroup
+    INNER JOIN hostgroups hg2
+        ON hg2.hostgroup_id = hg.hostgroup_id";
```

Comment:
Query will be broken if there is a search on hostgroup and servicegroup
```suggestion
        ON hg2.hostgroup_id = hg.hostgroup_id ";
```